caikit_nlp/toolkit/text_generation/tgis_utils.py

```diff
@@ -683,3 +687,33 @@ def unary_tokenize(
         return TokenizationResults(
             token_count=response.token_count,
         )
+
+
+def get_route_info(
+    context: Optional[RuntimeServerContextType],
+) -> Tuple[bool, Optional[str]]:
+    """
+    Returns a tuple `(True, x-route-info)` from context if "x-route-info" was found in
+    the headers/metadata.
+
+    Otherwise returns a tuple `(False, None)` if "x-route-info" was not found in the
+    context or if context is None.
+    """
```

Comment:
I think we can collapse the return into a simple `Optional[str]`. Returning `None` is indicative of "no route info found." This would simplify the logic that uses it and allow us to get another walrus!

Reply:
Done. Now using walrus operator for good @gabe-l-hart  karma.